tests/CMakeLists.txt

```
@@ -64,14 +64,15 @@ set_source_files_properties(${PROJECT_BINARY_DIR}/vk_safe_struct.cpp PROPERTIES
 add_executable(vk_extension_layer_tests
                extension_layer_tests.cpp
                synchronization2_tests.cpp
+               decompression_tests.cpp
                ${COMMON_CPP})
 add_test(NAME vk_extension_layer_tests COMMAND vk_extension_layer_tests)
 
 if(MSVC_IDE)
     set_target_properties(vk_extension_layer_tests PROPERTIES VS_DEBUGGER_ENVIRONMENT "VK_LAYER_PATH=$<TARGET_FILE_DIR:VkLayer_khronos_synchronization2>")
```

I see what is happening. You need both layers in VK_LAYER_PATH. On Windows use ; to separate the list.

For Visual Solutions you need to modify the `VS_DEBUGGER_ENVIRONMENT` property. The following should fix things up.

```suggestion
    set_target_properties(vk_extension_layer_tests PROPERTIES VS_DEBUGGER_ENVIRONMENT "VK_LAYER_PATH=$<TARGET_FILE_DIR:VkLayer_khronos_synchronization2>;$<TARGET_FILE_DIR:VkLayer_khronos_memory_decompression>")
```

---

I will add that. However, if an implementation supports "VK_NV_memory_decompression" extension then the layer disables itself. I think when you are running on NV GPU, since the driver supports this extension it is switching to the driver implementation. You will still need that env variable to force enable the layer instead.

I am assuming the test passed when you set VK_MEMORY_DECOMPRESSION_FORCE_ENABLE=true?
Since the point of these tests is to check the layer implementation, I think we should force this env variable to true.

Were these tests run on other GPUs(other than RTX 2080) and passed? I am wondering if you have an older NV driver on that system.

---

> I am assuming the test passed when you set VK_MEMORY_DECOMPRESSION_FORCE_ENABLE=true?

I don't have access to a WIN32 Nvidia machine at the moment. I'm making an inference based on the code in `tests/CMakeLists.txt`

You can also set `VK_MEMORY_DECOMPRESSION_FORCE_ENABLE` in the `VS_DEBUGGER_ENVIRONMENT` to force the layer instead.

---

Your suggestions only apply to when running through MSVC debugger environment or would it also take effect for the failure that we are talking about while running in the test environment?

Note that I have run these tests on several NV GPUs and one AMD GPU (RX 5700) and all these tests pass. So we are only debugging the failure that is reported here and I will appreciate debugging why it is failing on this particular system.  Adding debugger flags is probably not useful as my local debugging does not report failures.

I am going to push a change to vk_layer_settings.txt to force this env var = TRUE and hopefully it passes on that system.

---

Ignore this initial suggestion. It wouldn't affect the test failures since the Jenkins CI runs the raw executable and doesn't use ctest.